a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -879,3 +891,31 @@ def _mark_lb_as_error(self, vthunder, loadbalancers_list):
         except Exception as e:
             LOG.exception('Failed to set Loadbalancers to ERROR due to '
                           ': {}'.format(str(e)))
+
+
+class UpdateAcosVersionInVthunderEntry(VThunderBaseTask):
+
+    @axapi_client_decorator
+    def execute(self, vthunder, loadbalancer):
+        compute_flag = False
+        existing_vthunder = self.vthunder_repo.get_vthunder_by_project_id(db_apis.get_session(),
+                                                                          loadbalancer.project_id)
+        if existing_vthunder:
+            compute_flag = self.vthunder_repo.get_delete_compute_flag(db_apis.get_session(),
+                                                                      vthunder.compute_id)
+        if not existing_vthunder or compute_flag:
+            try:
+                acos_version_summary = self.axapi_client.system.action.get_acos_version()
+                acos_version = acos_version_summary['version']['oper']['sw-version'][:5]
```

Comment:
The version format is  [major].[minor].[patch]-[identifier]. And identifier is optional.
Can we keep them all?
[:5] is too risky, it may get wrong if major/minor number > 9

Reply:
Done
getting acos_version as [major].[minor].[patch]-[identifier] into db.
`acos_version = acos_version_summary['version']['oper']['sw-version'].split(',')[0]`